test/staking/RewardCalculation.test.ts

```diff
@@ -42,212 +43,255 @@ describe('Reward Calculation test', () => {
+    before(async () => {
+      snapshotId = await network.provider.send('evm_snapshot');
     });
 
-    it('Should be able to unstake and receives reward based on the smallest amount in the last period', async () => {
-      txs[0] = await stakingContract.unstake(userA.address, 250);
-      await expect(txs[0]).emit(stakingContract, 'UserRewardUpdated').withArgs(poolAddr, userA.address, 1600);
-      await expect(txs[0]).emit(stakingContract, 'PoolSharesUpdated').withArgs(period, poolAddr, 250);
-      expect(await stakingContract.getStakingAmount(poolAddr, userA.address)).eq(50);
-
-      txs[1] = await stakingContract.unstake(userB.address, 150);
-      await expect(txs[1]).emit(stakingContract, 'UserRewardUpdated').withArgs(poolAddr, userB.address, 400);
-      await expect(txs[1]).emit(stakingContract, 'PoolSharesUpdated').withArgs(period, poolAddr, 100);
-      expect(await stakingContract.getStakingAmount(poolAddr, userB.address)).eq(50);
-
-      aRps = aRps.add(MASK.mul(1000 / 100));
-      await stakingContract.increaseReward(1000);
-      tx = await stakingContract.endPeriod(); // period 5
-      await expect(tx)
-        .emit(stakingContract, 'PoolsUpdated')
-        .withArgs(period++, [poolAddr], [aRps], [await stakingContract.getStakingTotal(poolAddr)]);
-      expect(await stakingContract.getReward(poolAddr, userA.address)).eq(2100); // 50% of 1000 + 1600 from the last period
-      expect(await stakingContract.getReward(poolAddr, userB.address)).eq(900); // 50% of 1000 + 400 from the last period
+    after(async () => {
+      await network.provider.send('evm_revert', [snapshotId]);
     });
 
-    it('Should not distribute reward for the ones who unstake all in the period', async () => {
-      txs[1] = await stakingContract.unstake(userB.address, 50);
-      await expect(txs[1]).emit(stakingContract, 'PoolSharesUpdated').withArgs(period, poolAddr, 50);
-      await expect(txs[1]).emit(stakingContract, 'UserRewardUpdated').withArgs(poolAddr, userB.address, 900);
-      expect(await stakingContract.getStakingAmount(poolAddr, userB.address)).eq(0);
-
-      aRps = aRps.add(MASK.mul(1000 / 50));
+    it('Should be able to claim reward at the 1st period', async () => {
       await stakingContract.increaseReward(1000);
-      tx = await stakingContract.endPeriod(); // period 6
-      await expect(tx)
-        .emit(stakingContract, 'PoolsUpdated')
-        .withArgs(period++, [poolAddr], [aRps], [await stakingContract.getStakingTotal(poolAddr)]);
-      expect(await stakingContract.getReward(poolAddr, userA.address)).eq(3100); // 1000 + 2100 from the last period
+      tx = await stakingContract.endPeriod(); // period = 1
+      txs[0] = await stakingContract.claimReward(userA.address);
+      await expect(txs[0]).emit(stakingContract, 'RewardClaimed').withArgs(poolAddr, userA.address, 1000);
+      await expect(txs[0]).emit(stakingContract, 'UserRewardUpdated').withArgs(poolAddr, userA.address, 0);
     });
```

Comment:
Should add test of double claim / triple claim in the same period, and expect 0 amount of reward.